google/cloud/internal/source_accumulators.h

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_INTERNAL_SOURCE_ACCUMULATORS_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_INTERNAL_SOURCE_ACCUMULATORS_H
+
+#include "google/cloud/future.h"
+#include "google/cloud/version.h"
+#include "absl/meta/type_traits.h"
+#include "absl/types/variant.h"
+#include <vector>
+
+namespace google {
+namespace cloud {
+inline namespace GOOGLE_CLOUD_CPP_NS {
+namespace internal {
+
+template <typename Source>
+struct AccumulateAllEventsType {
+  using type = absl::variant<std::vector<typename Source::value_type>,
+                             typename Source::error_type>;
+};
+
+template <typename Source>
+using accumulate_all_events_t = typename AccumulateAllEventsType<Source>::type;
+
+/**
+ * Accumulate all the values generated by @p source into a vector.
+ *
+ * This is mostly used for testing, it uses the background threads to accumulate
+ * values, returning a `future<>` which is satisfied when the source is closed.
+ *
+ * The returned future is of type `future<variant<vector<T>, E>>` where `T` and
+ * `E` are the value type and error type in the original source. The returned
+ * future is satisfied with an error if the original source is closed with an
+ * error.
+ *
+ * @tparam Source
+ * @param source
+ * @return
+ */
+template <typename Source>
+future<accumulate_all_events_t<absl::decay_t<Source>>> accumulate_all(
```

Comment:
`accumulate_all()` doesn't match the function naming convention of our style guide. Is it intentionally different to match the style of something else?

Related question for `start()`, `on_next()` below.

Reply:
Fixed

> CamelCase delenda est